main.go

```diff
@@ -149,7 +149,7 @@ func registerEHR(server *echo.Echo, config Config, customerRepository customers.
 		log.Fatal(err)
 	}
 
-	fhirClientFactory := fhir.NewFactory(fhir.WithURL(config.FHIR.Server.Address))
+	fhirClientFactory := fhir.NewFactory(fhir.WithURL(config.FHIR.Server.Address), fhir.WithMultiTenancyEnabled(config.FHIR.Server.Type == "hapi"))
```

Comment:
does `Type == "hapi"` always imply multitenancy Or do we want to create an extra `hapi-multitenant` type?
Perhaps add a few FHIR Server constants like "generic" and "hapi"?

Reply:
That's what Rein used but it might be better to make it more explicit. Maybe adding `fhir.server.multiTenancyEnabled` in the configuration would be better?

Reply:
It depends if the tenant path is part of the broader fhir standard (i think not?), otherwise I would bind it to the hapi-multitenant server.

Reply:
IIRC it's only implemented for HAPI servers and not part of the FHIR spec